File: library/src/main/java/org/incendo/interfaces/drawable/Drawable.java
package org.incendo.interfaces.drawable;

import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.ItemStack;

@FunctionalInterface
public interface Drawable {

    static Drawable drawable(ItemStack itemStack) {
        return player -> itemStack;
    }

    static Drawable drawable(Material material) {
        return player -> new ItemStack(material);
    }

    ItemStack draw(Player player);
}

File: next/src/main/java/org/incendo/interfaces/next/drawable/Drawable.kt
package org.incendo.interfaces.next.drawable

import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack

public fun interface Drawable {

    public companion object {
        public fun drawable(item: ItemStack): Drawable = Drawable { item }

        public fun drawable(material: Material): Drawable = Drawable { ItemStack(material) }
    }

    public fun draw(player: Player): ItemStack
}
